### ISSUE TITLE
 Invalid argument: unknown flag --dev. See --help.

### ISSUE THREAD

**Issue by Mrthnx** [rest-nvim/rest.nvim#450]
### Prerequisites

- [X] I am using the latest stable release of Neovim
- [X] I am using the latest version of the plugin

### Neovim Version

v0.10.1

### Operating system/version

Ubuntu 20.04

### Actual behavior

        Error: Invalid argument: unknown flag --dev. See --help.
        Failed installing rest.nvim with `luarocks`.
![Screenshot_20240909_085958](https://github.com/user-attachments/assets/af306da8-801a-44d5-b86d-ee40805af466)

### Expected behavior

Correct installation of the plugin 

### Steps to reproduce

Instalacion del plugin atravez de lazyvim

### Repro (`lazy.nvim`)

```lua
vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

require("lazy.minit").repro({
    spec = {
        "rest-nvim/rest.nvim",
        {
            "nvim-treesitter/nvim-treesitter",
            build = ":TSUpdate",
            main = "nvim-treesitter.configs",
            opts = {
                ensure_installed = { "http" },
                sync_install = false,
                highlight = { enable = true },
                indent = { enable = true },
            },
        },
    },
})
```

**Comment by boltlessengineer** [comment]
Generally this seems to be a luarocks version issue. The plugin running `luarocks` to install `rest.nvim` is `lazy.nvim` and nto `rest.nvim` itself. I’m willing to help you though. Providing more informations can help me figuring out your issue.

- Can you reproduce it with `nvim -u repro.lua`? (with the reproduce lua snippet you put saved as `./repro.lua`)
Or are you having an issue only from your own config with [lazyvim](https://github.com/LazyVim/LazyVim)?
- Do you have `luarocks` installed? If so, what’s the version of it? (try run `luarocks --version`)

**Comment by Mrthnx** [comment]
Mi version de luarocks es /bin/luarocks 2.4.2.
veo que makefile del plugin tiene esos comandos

**Comment by boltlessengineer** [comment]
Yes we call luarocks in makefile but thats for testing and should not be run when your plugin manager install the plugin.
Can you reproduce with `nvim -u repro.lua`?

**Comment by Mrthnx** [comment]
I got these messages

`Error detected while processing /home/mrthnx/Developer/Personal/mini-http-nvim/repro.lua:
Failed to source `/home/mrthnx/Developer/Personal/mini-http-nvim/.repro/data/nvim/lazy/rest.nvim/plugin/rest-nvim.lua`
vim/_editor.lua:0: /home/mrthnx/Developer/Personal/mini-http-nvim/repro.lua..nvim_exec2() called at /home/mrthnx/Developer/Personal/mini-http-nvim/repro.lua:0../home/mrthnx/Developer/Personal/m
ini-http-nvim/.repro/data/nvim/lazy/rest.nvim/plugin/rest-nvim.lua: Vim(source):WARN: Dependency 'fidget.nvim' was not found. rest.nvim will be completely unable to show request progress messag
es
# stacktrace:
  - vim/_editor.lua:0 _in_ **cmd**
  - repro.lua:4`

![Screenshot_20240909_094042](https://github.com/user-attachments/assets/5ef06ad5-db5a-402e-b8e6-b044273ba0d1)

**Comment by boltlessengineer** [comment]
Seems like `lazy.nvim` don’t install `luarocks` via `hererocks` if `luarocks` is already installed. :thinking:
Then this seems to be a critical issue in `lazy.nvim`.

For quick fix, you can try upgrading the luarocks (at least `3.x`) installed in your OS or completely removing it (then `lazy.nvim` will install the latest automatically)

**Comment by Mrthnx** [comment]
was solved by deleting it and having lazy vim reinstall it.
